src/sql-lexer/src/keywords.rs

```diff
@@ -111,15 +120,16 @@ impl Keyword {
             // reserved prevents e.g. `SELECT pg_catalog.interval '1' year` from
             // parsing as `SELECT pg_catalog.interval '1' AS YEAR`.
             YEAR | MONTH | DAY | HOUR | MINUTE | SECOND
-        ) || self.is_reserved()
+        ) || self.is_always_reserved()
     }
 
     /// Reports whether a keyword is considered reserved in any context:
     /// either in table aliases, column aliases, or in all contexts.
     pub fn is_sometimes_reserved(self) -> bool {
-        self.is_reserved()
+        self.is_always_reserved()
             || self.is_reserved_in_table_alias()
             || self.is_reserved_in_column_alias()
+            || self.is_reserved()
```

Comment:
(Yes: we do need the new reservedness concept included here, to also fix the `SHOW CREATE VIEW` roundtripping bug, mentioned at the end of https://github.com/MaterializeInc/database-issues/issues/9968#issuecomment-3636123597)